TypeScript error when importing TypeScript

# Bug Report

### 🔎 Search Terms

LanguageServiceHost useCaseSensitiveFileNames

### 🕗 Version & Regression Information

<!-- When did you start seeing this bug occur?

"Bugs" that have existed in TS for a long time are very likely to be FAQs; refer to
  https://github.com/Microsoft/TypeScript/wiki/FAQ#common-bugs-that-arent-bugs

If possible, please try testing the nightly version of TS to see if it's already been fixed.
For npm: `typescript@next`
This is also the 'Nightly' version in the playground: http://www.typescriptlang.org/play/?ts=Nightly

Note: The TypeScript Playground can be used to try older versions of TypeScript.

Please keep and fill in the line that best applies:
-->
- This changed between versions 4.7.0-dev.20220311 and 4.7.0-dev.20220312

### ⏯ Playground Link

N/A, requires import statement which isn't possible in the playground

### 💻 Code

```ts
import * as ts from 'typescript';
```
Yep, that's it.

### 🙁 Actual behavior

The following error is generated:

```
node_modules/typescript/lib/typescript.d.ts:5716:15 - error TS2430: Interface 'LanguageServiceHost' incorrectly extends interface 'MinimalResolutionCacheHost'.  Types of property 'useCaseSensitiveFileNames' are incompatible.
    Type '(() => boolean) | undefined' is not assignable to type 'boolean | (() => boolean)'.
      Type 'undefined' is not assignable to type 'boolean | (() => boolean)'.

5716     interface LanguageServiceHost extends GetEffectiveTypeRootsHost, MinimalResolutionCacheHost {
                   ~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/typescript/lib/typescript.d.ts:5716
```

### 🙂 Expected behavior

TypeScript should compile without errors.

--- a/src/compiler/types.ts
+++ b/src/compiler/types.ts
@@ -6613,7 +6613,7 @@ namespace ts {
         realpath?(path: string): string;
         getCurrentDirectory?(): string;
         getDirectories?(path: string): string[];
-        useCaseSensitiveFileNames?: boolean | (() => boolean);
+        useCaseSensitiveFileNames?: boolean | (() => boolean) | undefined;
     }
 
     /**

--- a/tests/baselines/reference/api/tsserverlibrary.d.ts
+++ b/tests/baselines/reference/api/tsserverlibrary.d.ts
@@ -3164,7 +3164,7 @@ declare namespace ts {
         realpath?(path: string): string;
         getCurrentDirectory?(): string;
         getDirectories?(path: string): string[];
-        useCaseSensitiveFileNames?: boolean | (() => boolean);
+        useCaseSensitiveFileNames?: boolean | (() => boolean) | undefined;
     }
     /**
      * Used by services to specify the minimum host area required to set up source files under any compilation settings

--- a/tests/baselines/reference/api/typescript.d.ts
+++ b/tests/baselines/reference/api/typescript.d.ts
@@ -3164,7 +3164,7 @@ declare namespace ts {
         realpath?(path: string): string;
         getCurrentDirectory?(): string;
         getDirectories?(path: string): string[];
-        useCaseSensitiveFileNames?: boolean | (() => boolean);
+        useCaseSensitiveFileNames?: boolean | (() => boolean) | undefined;
     }
     /**
      * Used by services to specify the minimum host area required to set up source files under any compilation settings

